addon/components/array-container.js

```diff
@@ -310,11 +294,12 @@ export default Component.extend(PropTypeMixin, {
     onAddItem () {
       const bunsenId = this.get('bunsenId')
       const newItem = this._getEmptyItem()
+      const onChange = this.get('onChange')
       const value = this.get('value')
       const items = _.get(value, bunsenId) || []
       const index = items.length
 
-      this.notifyParentOfNewItem(newItem, index)
+      onChange(`${bunsenId}.${index}`, newItem)
```

Comment:
Matt's response, as discussed offline:

Yes b/c onChange propagates back up to the detail component, which in turn hands the change event off to the redux store (in bunsen-core), which ultimately updates the form value in the detail component